src/client/testing/testController/common/resultsHelper.ts

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as fsapi from 'fs-extra';
+import { TestItem, TestMessage, TestMessageSeverity, TestResultState, TestRun } from 'vscode';
+import { TestCase } from './testCase';
+import { TestCollection } from './testCollection';
+import { TestFile } from './testFile';
+import { TestFolder } from './testFolder';
+import { getTestCaseNodes } from './testItemUtilities';
+import { PythonTestData } from './types';
+
+type SupportedTestItemType = TestFolder | TestFile | TestCollection | TestCase;
+
+type TestSuiteResult = {
+    $: {
+        errors: string;
+        failures: string;
+        name: string;
+        skips: string;
+        skip: string;
+        tests: string;
+        time: string;
+    };
+    testcase: TestCaseResult[];
+};
+type TestCaseResult = {
+    $: {
+        classname: string;
+        file: string;
+        line: string;
+        name: string;
+        time: string;
+    };
+    failure: {
+        _: string;
+        $: { message: string; type: string };
+    }[];
+    error: {
+        _: string;
+        $: { message: string; type: string };
+    }[];
+    skipped: {
+        _: string;
+        $: { message: string; type: string };
+    }[];
+};
+
+async function parseXML(data: string): Promise<unknown> {
+    const xml2js = await import('xml2js');
+
+    return new Promise<unknown>((resolve, reject) => {
+        xml2js.parseString(data, (error: Error, result: unknown) => {
+            if (error) {
+                return reject(error);
+            }
+            return resolve(result);
+        });
+    });
+}
+
+function getJunitResults(parserResult: unknown): TestSuiteResult | undefined {
+    // This is the newer JUnit XML format (e.g. pytest 5.1 and later).
+    const fullResults = parserResult as { testsuites: { testsuite: TestSuiteResult[] } };
+    if (!fullResults.testsuites) {
+        return (parserResult as { testsuite: TestSuiteResult }).testsuite;
+    }
+
+    const junitSuites = fullResults.testsuites.testsuite;
+    if (!Array.isArray(junitSuites)) {
+        throw Error('bad JUnit XML data');
+    }
+    if (junitSuites.length === 0) {
+        return undefined;
+    }
+    if (junitSuites.length > 1) {
+        throw Error('got multiple XML results');
+    }
+    return junitSuites[0];
+}
+
+function getSafeInt(value: string, defaultValue = 0): number {
+    const num = parseInt(value, 10);
```

Comment:
```suggestion
    const num = Number.parseInt(value, 10);
```

This also applies to the other spots where `parseInt` is used directly, unless `parseInt` has something that `Number.parseInt` doesn't.